src/main/java/com/getindata/connectors/http/internal/table/lookup/AsyncHttpTableLookupFunction.java

```diff
@@ -73,29 +71,27 @@ public void open(FunctionContext context) throws Exception {
             );
     }
 
-    public void eval(CompletableFuture<Collection<RowData>> resultFuture, Object... keys) {
-
-        CompletableFuture<Optional<RowData>> future = new CompletableFuture<>();
-        future.completeAsync(() -> decorate.lookupByKeys(keys), pullingThreadPool);
+    @Override
+    public CompletableFuture<Collection<RowData>> asyncLookup(RowData keyRow) {
```

Comment:
@kristoffSC Please note that `AsyncLookupFunction#asyncLookup(RowData)` is called by `org.apache.flink.table.functions.AsyncLookupFunction#eval` which has the following implementation:

```java
    public final void eval(CompletableFuture<Collection<RowData>> future, Object... keys) {
        GenericRowData keyRow = GenericRowData.of(keys);
        asyncLookup(keyRow)
                .whenComplete(
                        (result, exception) -> {
                            if (exception != null) {
                                future.completeExceptionally(
                                        new TableException(
                                                String.format(
                                                        "Failed to asynchronously lookup entries with key '%s'",
                                                        keyRow),
                                                exception));
                                return;
                            }
                            future.complete(result);
                        });
    }
```

It is very similar to what we have in `asyncLookup` but it uses default thread pool while here we have `publishingThreadPool`.

Reply:
I dont think it uses default pool. From what I see it depends on the impl of `asyncLookup` so it is up to the implmenenbtation. The flink's `CachingAsyncLookupFunction` also delegated to `delegate` impl. The `whenComplete` will be executed on whatever thread was used to call `asyncLookup`

Reply:
You are right, thanks for double-checking it.